app/services/unified_price_service.py

```
@@ -262,21 +262,180 @@ async def _fetch_batch_from_source(self, symbols: List[str], source: PriceSource
+        except Exception as e:
+            self.logger.warning("Market data enterprise fallback failed", symbol=symbol, error=str(e))
+        
+        # 3. ENTERPRISE: Direct Exchange API calls
+        try:
+            from app.api.v1.endpoints.exchanges import get_binance_price, get_kucoin_prices
+            
+            # Binance ticker API
+            binance_price = await get_binance_price(symbol)
+            if binance_price and binance_price > 0:
+                sources_attempted.append("binance_direct")
+                self.logger.info(f"Binance direct fallback successful", symbol=symbol, price=binance_price)
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)